command/run.go

```diff
@@ -167,15 +172,34 @@ type Scanner struct {
 // NewScanner creates a new scanner with cancellation.
 func NewScanner(ctx context.Context, r io.Reader) *Scanner {
 	scanner := &Scanner{
-		ctx:     ctx,
-		Scanner: bufio.NewScanner(r),
-		linech:  make(chan string),
+		ctx:    ctx,
+		Reader: bufio.NewReader(r),
+		linech: make(chan string),
 	}
 
 	go scanner.scan()
 	return scanner
 }
 
+func (s *Scanner) readLine() (string, error) {
+	var (
+		isPrefix = true
+		err      error
+		line, ln []byte
+	)
+
+	// If the line was too long for the buffer then isPrefix is set and
+	// the beginning of the line is returned. The rest of the line will be
+	// returned from future calls. isPrefix will be false when returning
+	// the last fragment of the line.
+	for isPrefix && err == nil {
+		line, isPrefix, err = s.Reader.ReadLine()
```

Comment:
According to the [documentation](https://pkg.go.dev/bufio#Reader.ReadLine);

> ReadLine is a low-level line-reading primitive. Most callers should use ReadBytes('\n') or ReadString('\n') instead or use a Scanner.

Why do you prefer `ReadLine` over the recommended methods?

Reply:
Thanks.